Firestore/core/test/firebase/firestore/immutable/sorted_map_test.cc

```diff
@@ -110,6 +110,41 @@ TYPED_TEST(SortedMapTest, Increasing) {
     map = map.erase(i);
   }
   ASSERT_EQ(0u, map.size());
+
```

Comment:
Some notes on tests:
* do you plan on adding tests to `LlrbNodeIterator`?
* some test suggestions:
* tests showing that the iterators work well with standard algorithms (for example, any of `advance`, `distance`, `accumulate`, `binary_search`, `mismatch`, `copy_n`, `merge`).
* tests that show that when a tree is modified, existing iterators still iterate over the old version of the tree (assuming that's how they behave).

Reply:
These are tested in `sorted_map_test.cc`, and are effectively repeated for all of the iterators for `ArraySortedMap`, `TreeSortedMap`, and `SortedMap` through GoogleTest's `TYPED_TEST` mechanism.

I've added some of the tests you've suggested, though not all. For example, `std::binary_search` on these iterators would turn the O(lg(n)) `find` operation into O(n) because we don't need to implement these as random-access iterators.

Reply:
Thanks for adding those!